clients/rust/tests/initialize_config.rs

```diff
@@ -0,0 +1,581 @@
+                .minimum_balance(Config::LEN),
+            Config::LEN as u64,
+            &paladin_stake::ID,
+        );
+
+        let initialize_ix = InitializeConfigBuilder::new()
+            .config(config.pubkey())
+            .config_authority(authority)
+            .slash_authority(authority)
+            .mint(mint.pubkey())
+            .vault(token.pubkey())
+            .cooldown_time_seconds(1) // 1 second
+            .max_deactivation_basis_points(500) // 5%
+            .instruction();
+
+        // When we try to initialize the config with the wrong token authority.
+
+        let tx = Transaction::new_signed_with_payer(
+            &[create_mint_ix, create_ix, initialize_ix],
+            Some(&context.payer.pubkey()),
+            &[&context.payer, &mint, &config],
+            context.last_blockhash,
+        );
+
+        let err = context
+            .banks_client
+            .process_transaction(tx)
+            .await
+            .unwrap_err();
+
+        // Then we expect an error.
+
+        assert_instruction_error!(err, InstructionError::UninitializedAccount);
+    }
+
+    #[tokio::test]
+    async fn fail_initialize_config_with_wrong_account_length() {
+        let mut context = ProgramTest::new("stake_program", paladin_stake::ID, None)
+            .start_with_context()
+            .await;
+
+        // Given an empty config account and a mint.
+
+        let config = Keypair::new();
+        let authority = Keypair::new();
+        let authority_pubkey = authority.pubkey();
+
+        let mint = Keypair::new();
+        create_mint(
+            &mut context,
+            &mint,
+            &authority_pubkey,
+            Some(&authority_pubkey),
+            0,
+            MINT_EXTENSIONS,
+        )
+        .await
+        .unwrap();
+
+        let token = Keypair::new();
+        create_token(
+            &mut context,
+            &find_vault_pda(&config.pubkey()).0,
+            &token,
+            &mint.pubkey(),
+            TOKEN_ACCOUNT_EXTENSIONS,
+        )
+        .await
+        .unwrap();
+
+        let create_ix = system_instruction::create_account(
+            &context.payer.pubkey(),
+            &config.pubkey(),
+            context
+                .banks_client
+                .get_rent()
+                .await
+                .unwrap()
+                .minimum_balance(Config::LEN + 100),
+            (Config::LEN + 100) as u64, // <-- wrong length
+            &paladin_stake::ID,
+        );
+
+        let intialize_ix = InitializeConfigBuilder::new()
+            .config(config.pubkey())
+            .config_authority(authority_pubkey)
+            .slash_authority(authority_pubkey)
+            .mint(mint.pubkey())
+            .vault(token.pubkey())
+            .cooldown_time_seconds(1) // 1 second
+            .max_deactivation_basis_points(500) // 5%
+            .instruction();
+
+        // When we try to initialize the config with a non-empty token.
```

Comment:
micro-nit: Should probably say, "When we try to initialize the config with an incorrectly-sized account."

Reply:
Yep!